File: sol/647.cpp
struct Solution {
    int countSubstrings(string &s) {
        int n = s.size(), res = 0, i, j;
        for (int c = 0; c < n; ++c){
            i = j = c;
            while (i >= 0 && j < n && s[i] == s[j]){
                ++res;
                --i;
                ++j;
            }
            i = c, j = c + 1;
            while (i >= 0 && j < n && s[i] == s[j]){
                ++res;
                --i;
                ++j;
            }
        }
        return res;
    }
};

File: sol/647/647.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        n = len(s)
        res = 0
        for c in range(n):
            i, j = c, c
            while i >= 0 and j < n and s[i] == s[j]:
                res += 1
                i -= 1
                j += 1
            i, j = c, c + 1
            while i >= 0 and j < n and s[i] == s[j]:
                res += 1
                i -= 1
                j += 1
        return res
